ironfish/src/wallet/recentFeeCache.ts

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Assert } from '../assert'
+import { Blockchain } from '../blockchain'
+import { createRootLogger, Logger } from '../logger'
+import { PriorityQueue } from '../memPool'
+import { MetricsMonitor } from '../metrics'
+import { Block, Transaction } from '../primitives'
+import { Queue } from './queue'
+
+export class RecentFeeCache {
+  private queue: Queue<Transaction>
+  readonly chain: Blockchain
+  private readonly logger: Logger
+  private readonly metrics: MetricsMonitor
+  private numOfRecentBlocks = 10
+  private numOfTxSamples = 3
+  private defaultSuggestedFee = BigInt(2)
+
+  constructor(
+    chain: Blockchain,
+    recentBlocksNum?: number,
+    txSampleSize?: number,
+    metrics?: MetricsMonitor,
+    logger?: Logger,
+  ) {
+    this.logger = logger || createRootLogger()
+    this.numOfRecentBlocks = recentBlocksNum ?? this.numOfRecentBlocks
+    this.numOfTxSamples = txSampleSize ?? this.numOfTxSamples
+
+    this.queue = new Queue<Transaction>(this.numOfRecentBlocks * this.numOfTxSamples)
+    this.chain = chain
+    this.logger.withTag('recentFeeCache')
+    this.metrics = metrics || new MetricsMonitor({ logger: this.logger })
+  }
+
+  async setUpCache(): Promise<void> {
+    // Mempool is empty
+    let currentBlockHash = this.chain.latest.hash
+
+    for (let i = 0; i < this.numOfRecentBlocks; i++) {
+      const currentBlock = await this.chain.getBlock(currentBlockHash)
+      Assert.isNotNull(currentBlock, 'No block found')
+
+      const lowestFeeTransactions = this.getLowestFeeTransactions(currentBlock)
+      lowestFeeTransactions.forEach((tx) => {
+        if (this.queue.isFull()) {
+          return
+        }
+        this.queue.enqueue(tx)
+      })
+
+      currentBlockHash = currentBlock.header.previousBlockHash
+    }
+  }
+
+  /**
+   * Add recent transactions to fee cache
+   */
+
+  addFee(transaction: Transaction): boolean {
+    if (this.queue.isFull()) {
+      this.deleteOldestTransaction()
+    }
+    try {
+      this.queue.enqueue(transaction)
+      return true
+    } catch {
+      this.logger.error(`Error enqueue transaction to Recent Fee Cache`)
+      return false
+    }
```

Comment:
You don't want to do this type of try/catch in javascript. What error are you handling here? We should remove this try/catch.